rcl_action/src/rcl_action/goal_handle.c

```diff
@@ -0,0 +1,162 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl_action/goal_handle.h"
+
+#include "rcl/rcl.h"
+#include "rcl/error_handling.h"
+
+typedef struct rcl_action_goal_handle_impl_t
+{
+  rcl_action_goal_info_t * info;
```

Comment:
In retrospect, I guess there's no reason for this to be a pointer. I will refactor to `rcl_action_goal_info_t info;`

Reply:
Done d1fa431